openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -436,20 +436,18 @@ def create_library(
     )
     # Now create the library reference in our database:
     try:
-        # Atomic transaction required because if this fails,
-        # we need to delete the bundle in the exception handler.
-        with transaction.atomic():
-            ref = ContentLibrary.objects.create(
-                org=org,
-                slug=slug,
-                type=library_type,
-                bundle_uuid=bundle.uuid,
-                allow_public_learning=allow_public_learning,
-                allow_public_read=allow_public_read,
-                license=library_license,
-            )
+        ref = ContentLibrary.objects.create(
+            org=org,
+            slug=slug,
+            type=library_type,
+            bundle_uuid=bundle.uuid,
+            allow_public_learning=allow_public_learning,
+            allow_public_read=allow_public_read,
+            license=library_license,
+        )
     except IntegrityError:
-        delete_bundle(bundle.uuid)
+        # Note: the Content Library REST API view which calls this method is wrapped in an atomic transaction,
+        # so the bundle created above will be rolled back with the transaction.
```

Comment:
Yes, but isn't this whole file also a _python_ API which other python code is free to call whenever it wants, regardless of transaction? That could mean incorrect behavior if this python API is used outside of a transaction.
So if you are relying on an atomic transaction for correct behavior, why not check `transaction.get_autocommit()` at the start of this method and throw an exception if it's true, so that you can be sure it's only used in an atomic block?

Reply:
Good point -- fixed with [be40dd1](https://github.com/openedx/edx-platform/pull/30456/commits/be40dd12aa956546011ab72c5e15480dd1a9e5e0).